.github/workflows/ethereum-tests.yml

```diff
@@ -0,0 +1,35 @@
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
```

Comment:
Can you add `branches: [main]` under `pull_request`  so that is is executed on every pull request to `main`